bin/supervise.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+logger()
+{
+    level=${1:-info}
+    if [[ -n $JQ ]]; then
+        jq --raw-input --raw-output --compact-output \
+           --arg status "$level" \
+           '{
+                # theres no jq way I can find to format timestamp w/ subsecond precision
+                # so here we are, rolling our own!
+                timestamp: now |
+                           {
+                               # just the date-time stamp w/o Z
+                               datetime: . | strftime("%Y-%m-%dT%H:%M:%S"),
+                               # ms part of fractional seconds w/o leading 0
+                               ms: . | modf | .[0] | "\(.)" | .[1:5]
+                           } |
+                           "\(.datetime)\(.ms)Z",
+                status: $status,
+                message: .
+            }'
+    else
+        # provide a _very_ minimal fallback here
+        read message
+        echo "{\"status\":\"$level\",\"message\":\"$message\"}"
+    fi
+}
+
+# https://github.com/beacon-biosignals/K8sDeputy.jl/blob/b62e1858a4083ffc8f9f7b10fcb60a77896ae13e/src/graceful_termination.jl#L14
+IPC_DIR="${DEPUTY_IPC_DIR:-/run}"
+
+# https://github.com/beacon-biosignals/K8sDeputy.jl/blob/b62e1858a4083ffc8f9f7b10fcb60a77896ae13e/src/graceful_termination.jl#L21-L29
+get_pid()
+{
+    PID_FILE="${DEPUTY_IPC_DIR}/julia-entrypoint.pid"
```

Comment:
I would say Bash best practise is to use `local` instead of declaring global variables